### PR TITLE
Allow configurable sampling steps

### PR DESCRIPTION
**Pull request by brandonwillard**
This PR allows one to specify sampler functions that decide exactly how generated token sequences are sampled/produced from the logit values.  In doing so, this PR also adds a greedy "sampler" function (i.e. one that chooses the max logit values).

### CODE REVIEW

File: outlines/text/generate/sample.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    rng
+        A random number generator.
+
+    Returns
+    -------
+    The ids of the sampled tokens having shape ``(samples, n_seqs)``.
+
+    """
+    probs = torch.nn.functional.softmax(logits, dim=-1)
+    # next_token_ids = torch.multinomial(probs, num_samples=samples, generator=rng)
```
Comment (rlouf) [inline outlines/text/generate/sample.py:70]: `torch.multinomial` accepts matrices for `probs`: https://pytorch.org/docs/stable/generated/torch.multinomial.html

File: tests/text/generate/test_sequence.py [hunk 2]
```diff
@@ -67,38 +67,6 @@ def __hash__(self):
         return id(self)
 
 
-def test_vectorized_random_choice():
```
Comment (rlouf) [inline tests/text/generate/test_sequence.py:70]: Shouldn't this function still be tested?

Comment (brandonwillard) [reply tests/text/generate/test_sequence.py:70]: It was moved.

File: outlines/text/generate/sample.py (hunk 1 above)
Comment (brandonwillard) [reply outlines/text/generate/sample.py:70]: I have a few follow-up updates to add shortly; I'll put that in place alongside those.

Comment (brandonwillard) [reply outlines/text/generate/sample.py:70]: While `torch.multinomial` does work for matrices, we still need to check to the results against the higher dimensional tensor inputs in the tests (and update the tests that rely on RNG seeds), so the change should come in a follow-up.

**Comment by namin** [comment]
👍 :)

**Pull request merged**